routing/routes.go

```diff
@@ -2,33 +2,69 @@ package router
 	"github.com/ONSdigital/dp-apipoc-server/handler"
-	"github.com/julienschmidt/httprouter"
+	"github.com/ONSdigital/log.go/v2/log"
+	"github.com/gorilla/mux"
 )
 
 func GetRoutes(opsHandler *handler.OpsHandler, metaHandler *handler.MetadataHandler, dataHandler *handler.DataHandler) http.Handler {
-	router := httprouter.New()
+	router := mux.NewRouter()
 
-	router.HEAD("/ops/ping", opsHandler.PingHandler)
-	router.GET("/ops/status", opsHandler.StatusHandler)
+	router.HandleFunc("/ops/ping", opsHandler.PingHandler).Methods("HEAD")
+	router.HandleFunc("/ops/status", opsHandler.StatusHandler).Methods("GET")
 
-	router.GET("/dataset", metaHandler.GetDatasets)
-	router.GET("/timeseries", metaHandler.GetTimeseries)
+	router.HandleFunc("/dataset", metaHandler.GetDatasets).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}", metaHandler.GetSpecificDataset).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}/timeseries", metaHandler.GetSpecificDatasetTimeSeries).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}/timeseries/{timeseriesId}", metaHandler.GetSpecificTimeSeriesSpecificDataset).Methods("GET")
+	router.HandleFunc("/dataset/{datasetId}/timeseries/{timeseriesId}/data", dataHandler.GetData).Methods("GET")
 
-	router.GET("/dataset/:datasetId", metaHandler.GetSpecificDataset)
-	router.GET("/timeseries/:timeseriesId", metaHandler.GetSpecificTimeseries)
+	router.HandleFunc("/timeseries", metaHandler.GetTimeseries).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}", metaHandler.GetSpecificTimeseries).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}/dataset", metaHandler.GetSpecificTimeseriesDatasets).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}/dataset/{datasetId}", metaHandler.GetSpecificTimeSeriesSpecificDataset).Methods("GET")
+	router.HandleFunc("/timeseries/{timeseriesId}/dataset/{datasetId}/data", dataHandler.GetData).Methods("GET")
 
-	router.GET("/dataset/:datasetId/timeseries", metaHandler.GetSpecificDatasetTimeSeries)
-	router.GET("/timeseries/:timeseriesId/dataset", metaHandler.GetSpecificTimeseriesDatasets)
+	router.HandleFunc("/search", metaHandler.DoSearch).Methods("GET")
 
-	router.GET("/dataset/:datasetId/timeseries/:timeseriesId", metaHandler.GetSpecificTimeSeriesSpecificDataset)
-	router.GET("/timeseries/:timeseriesId/dataset/:datasetId", metaHandler.GetSpecificTimeSeriesSpecificDataset)
+	return router
+}
 
-	router.GET("/search", metaHandler.DoSearch)
+func DeprecationMiddleware(deprecation config.Deprecation) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 
-	router.GET("/dataset/:datasetId/timeseries/:timeseriesId/data", dataHandler.GetData)
-	router.GET("/timeseries/:timeseriesId/dataset/:datasetId/data", dataHandler.GetData)
+			if deprecation.Sunset != "" {
+				sunsetTime, err := time.Parse(time.RFC1123, deprecation.Sunset)
```

Comment:
strongly suggest moving this expensive parsing out of this function
to somewhere where it's done once (main?)

Reply:
I agree - I did think about this, so not sure why I didn't refactor